FormalConjectures/ErdosProblems/340.lean

```diff
@@ -0,0 +1,153 @@
+    let s' := greedySidon.go A.1 A.2 s
+    (⟨A ∪ {s'.1}, s'.2.2.2⟩, s')
+
+/-- `greedySidon` is the sequence obtained by the initial set $\{1\}$ and iteratively obtaining
+then next smallest integer that preserves the Sidon property of the set. This gives the
+sequence `1, 2, 4, 8, 13, 21, 31, ...`. -/
+def greedySidon (n : ℕ) : ℕ := greedySidon.aux n |>.2
+
+@[category test, AMS 5]
+example : greedySidon 0 = 1 := rfl
+
+@[category test, AMS 5]
+example : greedySidon 1 = 2 := by
+  decide
+
+@[category test, AMS 5]
+example : greedySidon 2 = 4 := by
+  decide
+
+@[category test, AMS 5]
+example : greedySidon 3 = 8 := by
+  decide
+@[category test, AMS 5]
+example : greedySidon 4 = 13 := by
+  decide
+
+@[category test, AMS 5]
+example : greedySidon 5 = 21 := by
+  decide
+
+@[category test, AMS 5]
+example : greedySidon 10 = 97 := by
+  decide +native
+
+/--
+Let $A = \{1, 2, 4, 8, 13, 21, 31, 45, 66, 81, 97, \ldots\}$ be the greedy Sidon sequence:
+we begin with $1$ and iteratively include the next smallest integer that preserves the
+Sidon property (i.e. there are no non-trivial solutions to $a + b = c + d$). What is the
+order of growth of $A$? Is it true that $| A \cap\{1, \ldots, N\}| \gg N^{1/2−\varepsilon}$
+for all $\varepsilon > 0$ and large $N$?
+-/
+@[category research open, AMS 5]
+theorem erdos_340 (ε : ℝ) (hε : ε > 0) :
+    (fun n : ℕ ↦ √n / n ^ ε) =O[atTop]
+      fun n : ℕ ↦ ((Set.range greedySidon ∩ Set.Iio n).ncard : ℝ) := by
+  sorry
+
+/--
+It is trivial that this sequence grows at least like $\gg N^{1/3}$.
+-/
+@[category undergraduate, AMS 5]
+theorem erdos_340.variants.third (ε : ℝ) (hε : ε > 0) :
+    (fun n : ℕ ↦ (n : ℝ) ^ ((1 : ℝ) / 3)) =O[atTop]
+      fun n : ℕ ↦ ((Set.range greedySidon ∩ Set.Iio n).ncard : ℝ) := by
+  sorry
+
+/--
+Erdős and Graham [ErGr80] also asked about the difference set $A - A$ and whether this has
+positive density.
+
+[ErGr80] Erdős, P. and Graham, R., Old and new problems and results in combinatorial number
+theory. Monographies de L'Enseignement Mathematique (1980).
+-/
+@[category research open, AMS 5]
+theorem erdos_340.variants.sub_hasPosDensity :
+    Set.HasPosDensity (Set.range greedySidon - Set.range greedySidon) :=
+  sorry
+
+/--
+Erdős and Graham [ErGr80] also asked about the difference set $A - A$ and whether this
+contains $22$, which it does.
+
+[ErGr80] Erdős, P. and Graham, R., Old and new problems and results in combinatorial number
+theory. Monographies de L'Enseignement Mathematique (1980).
+-/
+@[category research solved, AMS 5]
+theorem erdos_340.variants._22_mem_sub :
+    22 ∈ Set.range greedySidon - Set.range greedySidon := by
+  sorry
+
+/--
+The smallest integer which is unknown to be in $A - A$ is $33$.
+ -/
+@[category research open, AMS 5]
+theorem erdos_340.variants._33_mem_sub :
+    33 ∈ Set.range greedySidon - Set.range greedySidon ↔ answer(sorry) :=
+  sorry
+
+/--
+It may be true that all or almost all integers are in $A - A$.
```

Comment:
Would it be reasonable to add another related variant stating that the complement has density 0?